packages/payment/src/StripeService.ts

```diff
@@ -0,0 +1,105 @@
+import { injectable } from 'inversify';
+import { WishCardRepository } from '@donategifts/wishcard';
+import { UserRepository } from '@donategifts/user-data';
+import { logger } from '@donategifts/helper';
+import * as mongoSanitize from 'express-mongo-sanitize';
+import Stripe from 'stripe';
+import { IDonationHook, IStripeIntent, IUser, IWishCard, TypeObjectId } from '@donategifts/common';
+import { calculateWishItemTotalPrice, handleDonation } from './helper/paymentHelper';
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const stripe = new Stripe(process.env.STRIPE_KEY!, {
+	apiVersion: '2020-08-27',
+});
+
+@injectable()
+export class StripeService {
+	constructor(
+		private wishCardRepository: typeof WishCardRepository = WishCardRepository,
+		private lastWishcardDonation: string = '',
+		private userRepository: typeof UserRepository = UserRepository,
+	) {}
+
+	public async createIntent(
+		stripeData: IStripeIntent,
+	): Promise<{ success: boolean; error?: string; clientSecret?: string }> {
+		const { wishCardId, email, agencyName, userDonation, userId } = stripeData;
+		// Create a PaymentIntent with the order amount and currency
+		const wishCard = await this.wishCardRepository.getWishCardByObjectId(
+			mongoSanitize.sanitize(wishCardId) as TypeObjectId<IWishCard>,

```

Comment:
we shouldn't need to sanitize things here, if you still want to sanitize inputs i would do it in the controller and not the service. The service shouldn't need to sanitize typed parameters

you can also use ObjectId<IWishCard>(wishCardId) which returns the input as object id of the given type